app/src/main/java/com/epfl/drawyourpath/utils/Utils.kt

```diff
@@ -392,4 +407,15 @@ object Utils {
     fun coordinatesToBitmap(coordinates: List<LatLng>, size: Int = 100, paint: Paint = defaultPaint): Bitmap {
         return strokesToBitmap(listOf(coordinatesToStroke(coordinates)), size, paint)
     }
+
+    /**
+     * get the recognition result of a run
+     * @param run the run to recognize
+     * @return the recognition result
+     */
+    fun getRunRecognition(run: Run): CompletableFuture<RecognitionResult> {
+        return DigitalInk.downloadModelML().thenComposeAsync {
+            DigitalInk.recognizeDrawingML(coordinatesToInk(run.getPath().getPoints()), it)
```

Comment:
I think that saving the score in the cache must be done here, to evict the problem explained before.